pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -195,15 +202,45 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 		{
 			name: "No-op Running provision",
 			existing: []runtime.Object{
-				testClusterDeploymentWithProvision(),
+				func() *hivev1.ClusterDeployment {
+					cd := testClusterDeploymentWithProvision()
+					cd.Status.Conditions = append(cd.Status.Conditions, hivev1.ClusterDeploymentCondition{
+						Status:  corev1.ConditionFalse,
+						Type:    hivev1.InstallImagesNotResolvedCondition,
+						Reason:  imagesResolvedReason,
+						Message: imagesResolvedMsg,
+					})
+					cd.Status.Conditions = append(cd.Status.Conditions, hivev1.ClusterDeploymentCondition{
+						Status:  corev1.ConditionFalse,
+						Type:    hivev1.AuthenticationFailureClusterDeploymentCondition,
+						Reason:  platformAuthSuccessReason,
+						Message: "Platform credentials passed authentication check",
+					})
+					return cd
+				}(),
 				testProvision(),
 				testSecret(corev1.SecretTypeDockerConfigJson, pullSecretSecret, corev1.DockerConfigJsonKey, "{}"),
 				testSecret(corev1.SecretTypeDockerConfigJson, constants.GetMergedPullSecretName(testClusterDeployment()), corev1.DockerConfigJsonKey, "{}"),
 			},
 			validate: func(c client.Client, t *testing.T) {
 				cd := getCD(c)
 				if assert.NotNil(t, cd, "no clusterdeployment found") {
-					if e, a := testClusterDeploymentWithProvision(), cd; !assert.True(t, apiequality.Semantic.DeepEqual(e, a), "unexpected change in clusterdeployment") {
+					if e, a := func() *hivev1.ClusterDeployment {
+						cd := testClusterDeploymentWithProvision()
+						cd.Status.Conditions = append(cd.Status.Conditions, hivev1.ClusterDeploymentCondition{
+							Status:  corev1.ConditionFalse,
+							Type:    hivev1.InstallImagesNotResolvedCondition,
+							Reason:  imagesResolvedReason,
+							Message: imagesResolvedMsg,
+						})
+						cd.Status.Conditions = append(cd.Status.Conditions, hivev1.ClusterDeploymentCondition{
+							Status:  corev1.ConditionFalse,
+							Type:    hivev1.AuthenticationFailureClusterDeploymentCondition,
+							Reason:  platformAuthSuccessReason,
+							Message: "Platform credentials passed authentication check",
+						})
+						return cd
+					}(), cd; !assert.True(t, apiequality.Semantic.DeepEqual(e, a), "unexpected change in clusterdeployment") {
```

Comment:
It feels like there's a good assumed default state for test setup we could put into testClusterDeployment() function itself, ProvisionStopped=False, InstallImagesNotResolved=False, AuthenticationFailure=False. Does this work if you just assume to always set those "default" conditions in the default test input?

Reply:
That's a good idea - made a new function to do just that

Reply:
FYI did not change original testClusterDeployment() to allow for testing whether clusterdeployment conditions are initialized later on